approx/core/backend/_backend.py

```diff
@@ -26,30 +28,60 @@ def __init__(self, backend):
 class BackendEngine(abc.ABC):
     """Backend engine abstract class"""
 
-    def __init__(self):
-        self.type = None
-        self.module = None
+    def __init__(self) -> None:
+        super().__init__()
+        self.type: BackendType = BackendType.UNKNOWN
+        self.module: Any = None
 
-    def __str__(self):
-        if self.type:
-            return self.type.name.lower()
-        return "unknown"
+    def auto_quantize(self, model: Any, pretrained: bool) -> Any:
+        raise NotImplementedError("Need to implement this method")
+
+    def __str__(self) -> str:
+        return self.type.name.lower()
 
 
 class PyTorchBackend(BackendEngine):
     """PyTorch backend engine"""
 
-    def __init__(self):
+    def __init__(self) -> None:
+        super().__init__()
         import torch
 
         self.type = BackendType.TORCH
         self.module = torch
 
+    def auto_quantize(self, model: Any, pretrained: bool) -> Any:
+        """
+        Uses pytorch's dynamic quantization to convert some floating-point
+        `nn.Module` to `qint8`
+        Args:
+            model: The model to quantize.
+            pretrained: Whether the model is pretrained.
+
+        Returns:
+            The quantized model
+        """
+        if pretrained:
+            qmodel = self.module.quantization.quantize_dynamic(
+                model,
+                {
+                    # todo: this is definitely not enough
+                    self.module.nn.Linear,
+                    self.module.nn.Conv2d,
+                },
+                dtype=self.module.qint8,
```

Comment:
More quantization options?

@sudomaze

Reply:
I am adding more per [1](https://pytorch.org/docs/stable/_modules/torch/ao/quantization/quantize.html#quantize_dynamic) and [2](https://pytorch.org/docs/stable/quantization.html#quantization-customizations). Here is the current list that were added: 

```python
    self.module.nn.Linear,
    self.module.nn.LSTM,
    self.module.nn.GRU,
    self.module.nn.LSTMCell,
    self.module.nn.RNNCell,
    self.module.nn.GRUCell,
    self.module.nn.Conv2d,
    self.module.nn.Conv3d,
```

I will add some unit tests to ensure the quantization is working as expected. We will need to do a similar mapping in other backends.

Reply:
SGTM. I was mainly working on PyTorchBackend until we get that decently well working, before we support others